### PR TITLE
Switch to yield the select to the block

### PR DESCRIPTION
**Pull request by tpitale**
* accept select on x-option
* use select on x-option when register
* fallback to nearestNeighbor for non-ember-2 beta
* yield select to the block in handlebars
* pass the select for blockless version

### CODE REVIEW

**Comment by tpitale** [comment]
Attempt to resolve #44.

File: addon/components/x-option.js [hunk 1]
```diff
@@ -55,9 +64,9 @@ export default Ember.Component.extend({
     this._super.apply(this, arguments);
 
     Ember.run.scheduleOnce('afterRender', () => {
-      var select = this.nearestOfType(XSelectComponent);
+      /* Use an assigned select, or search for one nearby */
+      var select = this.get('select') || this.nearestOfType(XSelectComponent);
```
Comment (tpitale) [inline addon/components/x-option.js]: Still check for `nearestOfType` to try to avoid breaking anyone that hasn't upgrade to a version of ember that breaks it anyway.

**Comment by Robdel12** [comment]
I'm going to go ahead and close this since this bug is actually an ember bug. It _should_ be fixed in 1.13.4. If not we'll revisit :D

**Comment by tpitale** [comment]
This bugfix is gone in ember 2 beta. I believe it was intentionally removed. So, this will probably be broken again shortly.

This PR supports both the explicit yield, and the old syntax in 1.13.4. And it appears to be the way ember wants components to move.

**Comment by rwjblue** [comment]
> This bugfix is gone in ember 2 beta.

Can you point me to what you mean? `nearestOfType` should continue to work in 2.0.0-beta.3...

**Comment by Robdel12** [comment]
I'm also unsure how well the addon works in the beta right now. We need to upgrade ember-mocha to see where our test suite is at :)

**Comment by tpitale** [comment]
I'll try beta 3. It just appeared that the conversation was saying the code is or would soon be deprecated. I'm mostly concerned because I can't find the bug fix from 1.13.4 in master in `nearestOfType`.

**Comment by tpitale** [comment]
https://github.com/emberjs/ember.js/issues/11170 That conversation.

**Comment by cowboyd** [comment]
FWIW, I actually would prefer to not use `nearestOfType`, and once we can yield component helpers as block params it would be unecessary. Until then, it does provide the nicest api to the user.

**Comment by rwjblue** [comment]
From https://github.com/emberjs/ember.js/issues/11170#issuecomment-105060642:

> We decided not to deprecate accessing `parentView` at the moment because it does not make much sense to force component authors to switch to an alternative now only to switch to scoped helpers based solution when that lands (hopefully in an early 2.x release). 

**tldr;** what @cowboyd said :smiley_cat:

**Comment by tpitale** [comment]
@rwjblue Is that bug fix in Ember 2 beta? I couldn't find the code for it.

**Comment by rwjblue** [comment]
@tpitale - The actual code changes to fix the behavior has changed (done a different way now), but all the tests added in https://github.com/emberjs/ember.js/pull/11266 are still included and passing on beta and canary of Ember.

**Comment by tpitale** [comment]
Thanks @rwjblue. I'll try beta 3.
